### PR TITLE
Xan/sc 24385/tile server

### PR DESCRIPTION
**Pull request by XanthosXanthopoulos**
Currently we have not any proper data to test the server metadata and slice methods. I set this as a draft since there are no test implemented

### CODE REVIEW

File: tiledb/bioimg/utils/tiles_server.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+            for record in self._metadata
+        }
+
+        CORS(self._app)
+
+    def __del__(self) -> None:
+        self._pool.terminate()
+
+    def run(
+        self, host: str = "0.0.0.0", port: int = 8000, multiprocess: bool = False
```
Comment (gsakkis) [inline tiledb/bioimg/utils/tiles_server.py:222]: Generalize `multiprocess: bool = False` to `num_processes: int = 0`

File: tiledb/bioimg/utils/tiles_server.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    ) -> None:
+        print(platform.system())
```
Comment (gsakkis) [inline tiledb/bioimg/utils/tiles_server.py:224]: leftover print

File: tiledb/bioimg/utils/tiles_server.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        if platform.system() == "Windows":
+            self._pool.apply_async(self._app.run, kwds={"host": host, "port": port})
```
Comment (gsakkis) [inline tiledb/bioimg/utils/tiles_server.py:226]: What's wrong with `run_simple` on Windows?

**Review by gsakkis — CHANGES_REQUESTED** [review]
Looks great overall :100:  I left a few code style comments but nothing major.

Two more general comments:
- This doesn't have to be a package (also `utils` is usually a name for internal helpers). A plain `tiledb/bioimg/tiles_server.py` script will do.
- It should be packaged as an optional dependency and its dependencies (flask etc) be added to `extras_require` in setup.py.

File: tiledb/bioimg/utils/tiles_server.py [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+# from pathlib import Path
+from typing import Any, Callable, Dict, List, Tuple
+
+import cv2
+import flask
+import numpy as np
+from flask import Flask, Response, jsonify, make_response
+from flask_cors import CORS
+from multiprocess import Pool
```
Comment (gsakkis) [inline tiledb/bioimg/utils/tiles_server.py:15]: `s/multiprocess/multiprocessing`

**Comment by XanthosXanthopoulos** [comment]
The tile server will not be used for now so the PR closed

**Pull request closed without merging**